lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -823,15 +862,51 @@ def upload_reads(self, ctx, params):
         self.log('Starting upload reads, parsing args')
         o, wsid, name, objid, kbtype, single_end, fwdid, revid, shock = (
             self._proc_upload_reads_params(params))
-        interleaved = 1 if (not single_end and not revid) else 0
+        #IF from shock fwdid and revid are shock nodes, if not shock they are file paths
+        dfu = DataFileUtil(self.callback_url)
+        fwdpath = None
+        revpath = None
```

Comment:
I don't understand what you mean by declaration approach. In the current code the value fwdpath and revpath are always set later in the code, so initializing them here is pointless.